### ISSUE TITLE
Few small bugs in validation/testing (Change Detection - A2Net)

### ISSUE THREAD

**Issue by nikhilmakkar** [lwCVer/LWGANet#13]
1. Because of using the val dataloader the saved grid results overwrite the val images with test ones but still name them as val_xx_x.jpg
2. The 200 and 220 epoch numbers are confusing and more importantly they are being logged as val in tensorboard instead of test (which they really are) this is creating confusion too.

Small bugs but wanted to point out.

**Comment by lwCVer** [comment]
> 1. Because of using the val dataloader the saved grid results overwrite the val images with test ones but still name them as val_xx_x.jpg
> 2. The 200 and 220 epoch numbers are confusing and more importantly they are being logged as val in tensorboard instead of test (which they really are) this is creating confusion too.
> 
> Small bugs but wanted to point out.

200 and 220 respectively represent the weight results with the best precision in the validation set and the weights at the end of training. They are placed in the same tensorboard for a more intuitive comparison. Thank you for your feedback!

**Comment by nikhilmakkar** [comment]
Thanks!
